### PR TITLE
PNC fallback

### PR DESCRIPTION
**Pull request by phillip-kruger**
This PR allows you to configure artifacts that should be relocated.

### CODE REVIEW

**Comment by gabemontero** [comment]
/ok-to-test

but you've got some files out of date with main branch @phillip-kruger - you need to rebase

**Review by stuartwdouglas — APPROVED** [review]

File: java-components/sidecar/src/main/java/com/redhat/hacbs/sidecar/resources/HashUtil.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+    public static String sha1(byte[] value) {
+        try {
+            MessageDigest md = MessageDigest.getInstance("SHA-1");
+            byte[] digest = md.digest(value);
+            StringBuilder sb = new StringBuilder(40);
+            for (int i = 0; i < digest.length; ++i) {
+                sb.append(Integer.toHexString((digest[i] & 0xFF) | 0x100).substring(1, 3));
+            }
+            return sb.toString();
```
Comment (dwalluck) [inline java-components/sidecar/src/main/java/com/redhat/hacbs/sidecar/resources/HashUtil.java:24]: If we're using Java 17, it has a built in method for this now https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/HexFormat.html#formatHex(byte%5B%5D)

Comment (phillip-kruger) [reply java-components/sidecar/src/main/java/com/redhat/hacbs/sidecar/resources/HashUtil.java:24]: Ok, this was existing code that just moved to a Util class for reuse.

**Comment by gabemontero** [comment]
the jvm-build-service-e2e failure from yesterday was the known pruning timing issue

/test jvm-build-service-e2e

**Comment by gabemontero** [comment]
curious @stuartwdouglas @phillip-kruger - is configuring for artifacts to be relocated something that is self contained to the java side of things, or will a subsequent PR be coming to leverage this feature on the golang/operator stide?

thanks

**Comment by gabemontero** [comment]
using this PR to cross ref with Stuart's pruning fix

/test jvm-build-service-in-repo-with-pruning-e2e

**Comment by phillip-kruger** [comment]
> curious @stuartwdouglas @phillip-kruger - is configuring for artifacts to be relocated something that is self contained to the java side of things, or will a subsequent PR be coming to leverage this feature on the golang/operator stide?
> 
> thanks

I'll add to this PR (closing my draft PR) and also try and add the relevant e2e tests

**Comment by stuartwdouglas** [comment]
@gabemontero it should be limited to Java I think, as it happens in the sidecar. I don't think the operator really needs to know about it.

**Review by stuartwdouglas — APPROVED** [review]

**Review by stuartwdouglas — APPROVED** [review]

**Comment by stuartwdouglas** [comment]
Ignore the check sources failure, it is an issue with upstream KCP.

**Pull request merged**